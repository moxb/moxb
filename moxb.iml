<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager" inherit-compiler-output="true">
    <exclude-output />
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/packages/antd/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/meteor/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/moxb/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/react-html/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/examples/src" isTestSource="false" />
      <excludeFolder url="file://$MODULE_DIR$/packages/moxb/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/meteor/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/react-html/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/antd/dist" />
      <excludeFolder url="file://$MODULE_DIR$/admin/node-installation" />
      <excludeFolder url="file://$MODULE_DIR$/admin/bin-tools" />
      <excludeFolder url="file://$MODULE_DIR$/examples/dist" />
      <excludeFolder url="file://$MODULE_DIR$/examples/.cache" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>