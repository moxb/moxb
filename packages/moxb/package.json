{
  "name": "@moxb/moxb",
  "version": "0.2.0-beta.59",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "dist",
  "typings": "dist/",
  "files": [
    "dist/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ],
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest"
    },
    "testEnvironment": "node",
    "testMatch": [
      "**/src/**/__tests__/*.(ts|tsx)"
    ],
    "modulePaths": [
      "<rootDir>/peer-dependencies/node_modules"
    ]
  },
  "peerDependencies": {
    "mobx": "^5.15.6",
    "tslib": "^2.0.1"
  },
  "devDependencies": {
    "@types/atob": "2.1.2",
    "@types/jest": "26.0.13",
    "@types/lodash.flatten": "4.4.6",
    "@types/urijs": "1.19.11",
    "mobx": "5.13.1",
    "tslib": "1.10.0"
  },
  "dependencies": {
    "btoa-lite": "1.0.0",
    "history": "5.0.0",
    "lodash.flatten": "4.4.0",
    "moment": "2.29.1",
    "urijs": "1.19.2"
  }
}
