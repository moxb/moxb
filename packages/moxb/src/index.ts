export * from './impl';
export * from './types';
export * from './routing';
