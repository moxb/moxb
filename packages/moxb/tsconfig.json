{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["dom","es6","es2015","es2017", "dom.iterable","scripthost"],
        "inlineSourceMap": true,
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "./dist",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "allowUnreachableCode": false,
        "baseUrl": "./src",
        "paths": {
            "*": [
                "*",
                "../../../node_modules/*"
            ]
        }
    },
    "exclude": [
        "./node_modules",
        "./dist"
    ]
}
