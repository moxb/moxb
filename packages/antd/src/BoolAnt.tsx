import { Bool } from '@moxb/moxb';
import { Checkbox, Switch } from 'antd';
import { CheckboxProps } from 'antd/lib/checkbox';
import { SwitchSize } from 'antd/lib/switch';
import { FormItemProps } from 'antd/lib/form/FormItem';
import { observer } from 'mobx-react';
import * as React from 'react';
import { BindAntProps, labelWithHelp, parseProps } from './BindAnt';
import { FormItemAnt, parsePropsForChild } from './FormItemAnt';

@observer
export class BoolAnt extends React.Component<BindAntProps<Bool> & CheckboxProps> {
    render() {
        const { operation, invisible, children, label, ...props } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        // a null value renders the checkbox in intermediate state!
        const indeterminate = operation.value == null;
        return (
            <Checkbox
                data-testid={operation.id}
                checked={operation.value}
                onChange={() => operation.toggle()}
                indeterminate={indeterminate}
                {...props}
            >
                {children}
                {labelWithHelp(label, operation.help, operation.id)}
            </Checkbox>
        );
    }
}

export interface AllowedSwitchProps {
    prefixCls?: string;
    size?: SwitchSize;
    className?: string;
    checkedChildren?: React.ReactNode;
    unCheckedChildren?: React.ReactNode;
    autoFocus?: boolean;
    style?: React.CSSProperties;
}

@observer
export class BoolSwitchAnt extends React.Component<BindAntProps<Bool> & AllowedSwitchProps> {
    render() {
        const { operation, invisible, children, label, reason, disabled, ...props } = parseProps(
            this.props,
            this.props.operation
        );
        if (invisible) {
            return null;
        }
        const switchClassName = disabled ? 'ant-checkbox-disabled' : '';
        const labelClassName = disabled ? '' : 'ant-checkbox-wrapper';
        return (
            <span title={reason}>
                <span className={switchClassName}>
                    <Switch
                        disabled={disabled}
                        data-testid={operation.id}
                        checked={operation.value}
                        onChange={() => operation.toggle()}
                        {...props}
                    />
                </span>
                &nbsp; &nbsp;
                <span onClick={() => operation.toggle()} className={labelClassName}>
                    {labelWithHelp(label, operation.help, operation.id)}
                </span>
            </span>
        );
    }
}

@observer
export class BoolFormAnt extends React.Component<BindAntProps<Bool> & Partial<FormItemProps> & CheckboxProps> {
    render() {
        const { operation, invisible, ...props } = parsePropsForChild(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <FormItemAnt operation={operation} {...(this.props as any)}>
                <BoolAnt operation={operation} {...props} />
            </FormItemAnt>
        );
    }
}

@observer
export class BoolSwitchFormAnt extends React.Component<
    BindAntProps<Bool> & Partial<FormItemProps> & AllowedSwitchProps
> {
    render() {
        const { operation, invisible, ...props } = parsePropsForChild(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <FormItemAnt operation={operation} {...(this.props as any)}>
                <BoolSwitchAnt operation={operation} {...props} />
            </FormItemAnt>
        );
    }
}
