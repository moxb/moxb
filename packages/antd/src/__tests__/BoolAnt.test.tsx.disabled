import { BoolImpl } from '@moxb/moxb';
import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { BoolAnt, BoolFormAnt } from '../BoolAnt';
import { shallowMoxbToJson } from './enzymeHelper';

describe('BoolAnt', function () {
    it('should render a checkbox by default', function () {
        const operation = new BoolImpl({
            id: 'TheId',
            label: 'oh',
        });
        expect(shallowMoxbToJson(shallow(<BoolAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should return null if invisible', function () {
        const operation = new BoolImpl({
            id: 'TheId',
            label: 'oh',
            invisible: () => true,
        });
        expect(shallow(<BoolAnt operation={operation} />).type()).toBeNull();
    });

    it('should call onChange if the input value changes', function () {
        const operation = new BoolImpl({
            id: 'TheId',
            label: 'oh',
        });
        const wrapper = mount(<BoolAnt operation={operation} />);
        const input: any = wrapper.find('input').first();
        input.simulate('change', { target: { checked: true } });
        expect(wrapper.find('input').first().prop('checked')).toBe(true);
    });
});

describe('BoolFormAnt', function () {
    it('should render a checkbox by default', function () {
        const operation = new BoolImpl({
            id: 'TheId',
            label: 'oh',
        });
        expect(shallowMoxbToJson(shallow(<BoolFormAnt operation={operation} />))).toMatchSnapshot();
    });

    it('should return null if invisible', function () {
        const operation = new BoolImpl({
            id: 'TheId',
            label: 'oh',
            invisible: () => true,
        });
        expect(shallow(<BoolFormAnt operation={operation} />).type()).toBeNull();
    });
});
