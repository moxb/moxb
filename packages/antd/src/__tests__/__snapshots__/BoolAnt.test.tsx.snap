// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoolAnt should render a checkbox by default 1`] = `
<span>
  <Checkbox
    data-testid="TheId"
    disabled={false}
    id="the_id"
    indeterminate={true}
    onChange={[Function]}
    readOnly={false}
  >
    oh
  </Checkbox>
</span>
`;

exports[`BoolFormAnt should render a checkbox by default 1`] = `
<FormItemAnt
  operation="bound BoolImpl"
>
  <BoolAnt
    disabled={false}
    id="the_id"
    operation="bound BoolImpl"
    readOnly={false}
  />
</FormItemAnt>
`;
